apps/sms/views/conversation/js/conversation.js

```diff
@@ -2912,103 +2890,62 @@ var ConversationView = {
     new OptionMenu(params).show();
   },
 
-  onCreateContact: function conv_onCreateContact() {
-    InboxView.updateContactsInfo();
-    // Update Header if needed
-    if (Navigation.isCurrentPanel('thread')) {
-      ConversationView.updateHeaderData();
+  discardDraft: function conv_discardDraft() {
+    // If we were tracking a draft properly, update the Drafts object entry.
+    if (!this.draft) {
+      return;
     }
-  },
 
-  discardDraft: function conv_discardDraft() {
-    // If we were tracking a draft
-    // properly update the Drafts object
-    // and InboxView entries
-    if (this.draft) {
-      Drafts.delete(this.draft).store();
-      if (Threads.active) {
-        Threads.active.timestamp = Date.now();
-        InboxView.updateThread(Threads.active);
-      } else {
-        InboxView.removeThread(this.draft.id);
-      }
-      this.draft = null;
+    // Update thread timestamp. Will be removed in bug 958105.
+    var thread = Threads.active;
+    if (thread) {
+      thread.timestamp = Date.now();
     }
+
+    Drafts.delete(this.draft).store();
+
+    this.draft = null;
   },
 
-   /**
-   * saveDraft
-   *
-   * Saves the currently unsent message content or recipients
-   * into a Draft object.  Preserves the currently marked
-   * draft if specified.  Draft preservation is intended to
-   * keep this.draft populated with the currently
-   * showing draft when the app is hidden, so when the app
-   * comes out of hiding, it knows there is a draft to continue
-   * to keep track of.
+  /**
+   * Saves currently unsent message content and/or recipients into a Draft
+   * object. Preserves currently active draft if specified. Draft preservation
+   * is intended to keep this.draft populated with the currently active draft
+   * when app is hidden, so that when app becomes visible again, it knows there
+   * is a draft to continue to keep track of.
    *
-   * @param {Object} opts Optional parameters for saving a draft.
-   *                  - preserve, boolean whether or not to preserve draft.
-   *                  - autoSave, boolean whether this is an auto save.
+   * @param {boolean?} preserveDraft Optional boolean parameter that indicates
+   * whether or not we should preserve draft.
    */
-  saveDraft: function conv_saveDraft(opts) {
-    var content, draft, recipients, subject, thread, threadId, type;
-
-    content = Compose.getContent();
-    subject = Compose.getSubject();
-    type = Compose.type;
-
-    if (Threads.active) {
-      recipients = Threads.active.participants;
-      threadId = Threads.currentId;
-    } else {
-      recipients = this.recipients.numbers;
-    }
+  saveDraft: function conv_saveDraft(preserveDraft) {
+    var thread = Threads.active;
 
-    var draftId = this.draft ? this.draft.id : null;
+    // Do we need to save participants for thread draft?
```

Comment:
maybe not, but saving the recipients for thread draft for keeping the same interface seem not a bad decision anyway